libfwupdplugin/fu-common.c

```diff
@@ -838,10 +838,14 @@ fu_common_spawn_sync(const gchar *const *argv,
 	fu_common_spawn_create_pollable_source(helper);
 	g_main_loop_run(helper->loop);
 	g_cancellable_disconnect(cancellable, cancellable_id);
-	if (g_cancellable_set_error_if_cancelled(helper->cancellable, error))
-		return FALSE;
 #endif
-	return g_subprocess_wait_check(subprocess, cancellable, error);
+	if (!g_subprocess_wait_check(subprocess, cancellable, error))
+		return FALSE;
+	if (g_cancellable_set_error_if_cancelled(cancellable, error))
```

Comment:
helper->cancellable

Reply:
Correct -- I told you it was late! Can you try what I just pushed please.

Reply:
The test does not leak processes anymore! Thanks!